<a href="https://colab.research.google.com/github/julianamu323/Proyectos/blob/main/Afecciones_cardiacas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.formula.api as smf
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
datos = pd.read_csv('/content/drive/MyDrive/Proyectos/Datos/heart_attack_youth_adult_france.csv',
                   sep = ',',
                   low_memory = False)
datos

Patient_ID  Age     Sex  Weight_kg  Height_cm   BMI  \
0                1   56    Male      118.4        184  35.4   
1                2   46    Male       50.7        164  32.6   
2                3   32    Male      118.0        179  39.8   
3                4   60    Male       96.8        161  23.7   
4                5   25    Male       53.9        169  24.4   
...            ...  ...     ...        ...        ...   ...   
266781      266782   33    Male      108.3        160  35.9   
266782      266783   33  Female       75.1        186  19.8   
266783      266784   27  Female       56.5        198  34.7   
266784      266785   26    Male       70.2        180  35.8   
266785      266786   43  Female       61.6        173  37.8   

        Blood_Pressure_Systolic  Blood_Pressure_Diastolic  Cholesterol_Level  \
0                            96                        94              204.3   
1                           114                        65              279.4   
2                            98                       118              291.4   
3                           108                       101              273.3   
4                           141                        73              290.8   
...                         ...                       ...                ...   
266781                      121                       102              152.7   
266782                      164                       119              201.8   
266783                      159                        74              160.9   
266784                      138                       112              270.3   
266785                       99                       100              223.9   

        Smoking_Status  ...  Exercise_Induced_Pain  Heart_Attack Age_Group  \
0            Ex-Smoker  ...                     No            No     Adult   
1           Non-Smoker  ...                     No           Yes     Adult   
2       Current Smoker  ...                     No            No     Adult   
3           Non-Smoker  ...                     No            No     Youth   
4            Ex-Smoker  ...                     No            No     Youth   
...                ...  ...                    ...           ...       ...   
266781       Ex-Smoker  ...                     No           Yes     Youth   
266782      Non-Smoker  ...                     No           Yes     Youth   
266783      Non-Smoker  ...                     No            No     Adult   
266784  Current Smoker  ...                     No            No     Adult   
266785      Non-Smoker  ...                     No            No     Adult   

         Region Air_Pollution_Level  Income_Level  Education_Level  \
0       Central                  67        Medium          College   
1         North                  16           Low      High School   
2         South                  32        Medium          College   
3          East                  86           Low          College   
4         South                  15           Low      High School   
...         ...                 ...           ...              ...   
266781    North                  16          High      High School   
266782    South                   5           Low          College   
266783     West                  55        Medium         Graduate   
266784  Central                  38           Low          College   
266785    North                  51          High          College   

       Health_Insurance Regular_Checkups Medication_Adherence  
0                   Yes     Occasionally                  Yes  
1                   Yes     Occasionally                  Yes  
2                   Yes       Frequently                  Yes  
3                   Yes     Occasionally                  Yes  
4                   Yes     Occasionally                   No  
...                 ...              ...                  ...  
266781              Yes     Occasionally                  Yes  


In [4]:
data = datos.copy()

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266786 entries, 0 to 266785
Data columns (total 27 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   Patient_ID                266786 non-null  int64  
 1   Age                       266786 non-null  int64  
 2   Sex                       266786 non-null  object 
 3   Weight_kg                 266786 non-null  float64
 4   Height_cm                 266786 non-null  int64  
 5   BMI                       266786 non-null  float64
 6   Blood_Pressure_Systolic   266786 non-null  int64  
 7   Blood_Pressure_Diastolic  266786 non-null  int64  
 8   Cholesterol_Level         266786 non-null  float64
 9   Smoking_Status            266786 non-null  object 
 10  Alcohol_Consumption       266786 non-null  int64  
 11  Physical_Activity_Hours   266786 non-null  float64
 12  Diabetes                  266786 non-null  object 
 13  Family_History            266786 non-null  o

### **Descripción de las variables.**

Dentro del conjunto de datos podemos encontrar variables de tipo cuantitativo y cualitativo las cuales se describen a continuación:

#### **Variables cuantitativas:**
- **Patient ID:** Identificación del paciente (1 - 10.000).
- **Age:** Edad (18 años - 60 años).
- **Weight_kg:** Peso medido en kilogramos (kg).
- **Height_cm:** Altura medida en centímetros (cm).
- **BMI:** Índice de masa corporal.
- **Blood Pressure Systolic:** Presión Arterial Sistólica (90 - 179).
- **Blood Pressure Diastolic:** Presión Arterial Diastólica (60 - 119).
- **Cholesterol Level:** Nivel de colesterol (150.0 - 300.0).
- **Alcohol Consumption:** Consumo de alcohol (0 - 19).
- **Physical Activity hours:** Horas de actividad física (0.0 - 9.9).
- **Stress level:** Nivel de estrés (1 - 10).
- **Heart rate:** Frecuencia cardiaca.
- **Air Pollution level:** Nivel de contaminación del aire (1 - 100).

#### **Variables cualitativas:**
- **Sex:** Sexo (Female, Male y other).
- **Smoking Status:** Estado de fumador (Ex-Smoker, Non-Smoker y Current-Smoker).
- **Diabetes:** Diabetes (Yes - No).
- **Family History:** Historia familiar (Yes - No).
- **Diet type:** Tipo de dieta (Healthy, Mixed y Unheralthy).
- **Excersice Induced Pain:** Dolor inducido por el ejercicio (Yes - No).
- **Heart attack:** Ataque al corazón (Yes - No).
- **Age group:** Grupo o rango de edad (Adulth - Youth).
- **Region:** Región (norte, sur, centro, oeste y este de Francia).
- **Income level:** Nivel de ingresos (High, Low, Medium).
- **Education level:** Nivel de educación (College, Graduate, High School).
- **Health Insurance:** Seguros de salud (Yes - No).
- **Regular Checkups:** Chequeos regulares (Frequently, Occasionally, Rarely).
- **Medication adherence:** Adherencia a la medicación (Yes - No).

Con el fin de mejorar la comprensión de la base de datos, se opta por nombrar algunas de las variables de estudio:

- **Weight_kg:** Weight.
- **Height_cm:** Height.
- **Blood Pressure Sytolic:** BPS.
- **Blood Pressure Diastolic:** BPD.
- **Cholesterol level:** Cholesterol.
- **Physical Activity Hours:** PAH.
- **Excersice Induced Pain:** EIP.

In [10]:
data.rename(columns={'Weight_kg':'Weight', 'Height_cm':'Height',
                     'Blood_Pressure_Systolic':'BPS',
                     'Blood_Pressure_Diastolic':'BPD',
                     'Cholesterol_Level':'Cholesterol',
                     'Physical_Activity_Hours':'PAH',
                     'Exercise_Induced_Pain':'EIP'}, inplace=True)